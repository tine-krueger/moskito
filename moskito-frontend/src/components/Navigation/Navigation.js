import styled from 'styled-components/macro'
import { Link } from 'react-router-dom'
import { FiLogOut } from 'react-icons/fi'
import { AiOutlineSearch, AiFillPushpin } from 'react-icons/ai'
import { useAuth } from '../../context/auth'



export default function Navigation() {
    const { deleteTokens, setAuthTokens, authTokens } = useAuth()
    return (
        <Navbar>
            <NavList>
                <NavListItem>
                    <Link to="/find-campsite"><AiOutlineSearch /></Link>
                </NavListItem>
                <NavListItem data-testid='bookmarks'>
                    <Link to="/bookmarks"><AiFillPushpin /></Link>
                </NavListItem>
                <NavListItem>
                    <button onClick={logOut}><FiLogOut /></button>
                </NavListItem>
            </NavList>
        </Navbar>
    )

    function logOut(event){
        event.preventDefault()
        const token = authTokens
        setAuthTokens()
        deleteTokens(token)
    }
}

const Navbar = styled.nav`
    margin: 0 auto;
    max-width: 500px;
    background-color: var(--background);
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
    box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6), -2px -2px 2px 0px rgba(255,255,255,0.5);
`

const NavList = styled.ul`
    padding-left:0;
    list-style-type: none;
    display: flex;
    justify-content: space-around;
`    
const NavListItem = styled.li`
    text-align: center;
    border-radius: 10px;
    box-shadow: 2px 4px 8px -4px rgba(0, 0, 0, 0.6), 
                -1px -1px 2px 0px rgba(255, 255, 255, 0.5);
    width: 2.3em;
    height: 2.3em;
    display: flex;
    justify-content: center;
    align-items: center;

    button{
        background-color: transparent;
        border: none;
        
    }
    button:hover {
        outline: none;
    }
    a svg, button svg {
        width: 1.5em;
        height: 1.5em;
        stroke: var(--link);
    }
    button svg{
        width: 1.7em;
        height: 1.7em;
    }

    &:hover,
    &:active,
    &:focus {
        box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2), 
                inset -3px -4px 6px 0 rgba(255,255,255,0.3);
        outline: none;
    }
` 
