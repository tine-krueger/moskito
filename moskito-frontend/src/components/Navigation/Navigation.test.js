import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import 'jest-styled-components'
import { MemoryRouter } from 'react-router-dom'
import Navigation from './Navigation'

const mockSetAuthTokens = jest.fn()
const mockGetToken = jest.fn()

jest.mock('../../context/auth', () => ({
    ...jest.requireActual('../../context/auth'),
    useAuth: () => ({
        setAuthTokens: mockSetAuthTokens,
        getToken: mockGetToken
    }),
}))

describe('Navigation', () => {

    it('renders correctly', () => {
        const { container } = render(
        <MemoryRouter>
            <Navigation/>
        </MemoryRouter>
        )
        expect(container).toMatchSnapshot()
    })

    it('Nav-Link changes style when hovered', () => {
        const { getByTestId } = render(
            <MemoryRouter>
                <Navigation/>
            </MemoryRouter>
            )
        
        const bookmarksLink = getByTestId('bookmarks')
        userEvent.hover(bookmarksLink)
        
        expect(bookmarksLink).toHaveStyle('background: linear-gradient(145deg, #A2B2B0, #E2FAF6)')
    })

})