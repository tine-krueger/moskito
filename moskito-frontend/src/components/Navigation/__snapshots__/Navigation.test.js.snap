// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
.c0 {
  margin: 0 auto;
  max-width: 500px;
  background-color: var(--background);
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
}

.c1 {
  padding-left: 0;
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c2 {
  text-align: center;
  border-radius: 10px;
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
  width: 2.3em;
  height: 2.3em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 button {
  background-color: transparent;
  border: none;
}

.c2 button:hover {
  outline: none;
}

.c2 a svg,
.c2 button svg {
  width: 1.5em;
  height: 1.5em;
  stroke: var(--link);
}

.c2 button svg {
  width: 1.7em;
  height: 1.7em;
}

.c2:hover,
.c2:active,
.c2:focus {
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  outline: none;
}

<div>
  <nav
    class="c0"
  >
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <a
          href="/find-campsite"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </a>
      </li>
      <li
        class="c2"
      >
        <a
          data-testid="bookmarks"
          href="/bookmarks"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M878.3 392.1L631.9 145.7c-6.5-6.5-15-9.7-23.5-9.7s-17 3.2-23.5 9.7L423.8 306.9c-12.2-1.4-24.5-2-36.8-2-73.2 0-146.4 24.1-206.5 72.3-15.4 12.3-16.6 35.4-2.7 49.4l181.7 181.7-215.4 215.2a15.8 15.8 0 0 0-4.6 9.8l-3.4 37.2c-.9 9.4 6.6 17.4 15.9 17.4.5 0 1 0 1.5-.1l37.2-3.4c3.7-.3 7.2-2 9.8-4.6l215.4-215.4 181.7 181.7c6.5 6.5 15 9.7 23.5 9.7 9.7 0 19.3-4.2 25.9-12.4 56.3-70.3 79.7-158.3 70.2-243.4l161.1-161.1c12.9-12.8 12.9-33.8 0-46.8z"
            />
          </svg>
        </a>
      </li>
      <li
        class="c2"
      >
        <button
          data-testid="logout"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
